harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -0,0 +1,195 @@
+            input_path = dataset.filepath()
+
+            if is_variable_in_dataset(dimension_path, dataset):
+                dim_data = self.input_dimensions.setdefault(dimension_path, {})
+
+                if input_path not in dim_data:
+                    dim_data[input_path] = DimensionInformation(dataset,
+                                                                dimension_path)
+
+
+class DimensionInformation:
+    """ A class containing information on a dimension, including the path of
+        the dimension variable within a NetCDF-4 or Zarr Dataset, the values of
+        the 1-dimensional dimension array and any associated temporal epoch
+        and unit. For TRT-121, it is initially assumed that non-temporal
+        dimensions can be aggregated without requiring offsets similar to
+        that stored in the temporal dimensions `units` metdata attribute.
+
+        This class can be used to contain both the dimension information from
+        individual input NetCDF-4 files, as well as the aggregated output
+        dimension.
+
+    """
+    def __init__(self, dataset: Dataset, dimension_path: str):
+        self.dimension_path = dimension_path
+        self.values = dataset[dimension_path][:]
+        self.units = get_nc_attribute(dataset[dimension_path], 'units')
+        self.epoch = None
+        self.time_unit = None
+        self._get_epoch_and_unit()
+
+    def _get_epoch_and_unit(self):
+        """ Check the `units` attribute in the dimension variable metadata. If
+            present, compare the format to the CF-Convention format for a
+            temporal dimension (e.g., "seconds since 2000-01-02T03:04:05"), and
+            extract the type of unit.
+
+            For documentation on temporal dimensions see:
+
+            https://cfconventions.org/cf-conventions/cf-conventions.html#time-coordinate
+
+        """
+        if self.units is not None and ' since ' in self.units:
+            time_unit_string, epoch_string = self.units.split(' since ')
+            self.epoch = parse_datetime(epoch_string)
+            self.time_unit = time_unit_to_delta_map.get(time_unit_string)
+
+
+def get_nc_attribute(
+    variable: Variable, attribute_name: str,
+    default_value: Optional[NetCDF4Attribute] = None
+) -> Optional[NetCDF4Attribute]:
+    """ A helper function that attempts to retrieve the value of a metadata
+        attribute. If that attribute is missing the resulting `AttributeError`
+        is handled and an optional default value is applied.
+
+    """
+    try:
+        nc_attribute = variable.getncattr(attribute_name)
+    except AttributeError:
+        nc_attribute = default_value
+
+    return nc_attribute
+
+
+def resolve_reference_path(variable: Variable, reference: str) -> str:
+    """ Extract the full path of a dimension reference based upon the path of
+        the variable containing the reference.
+
+        If the reference has a leading slash, it is assumedto already be
```

Comment:
assumedto -> assumed to

Reply:
Ha - I _always_ manage to mistype at least one comment. Good catch.